scene.cpp

```diff
@@ -23,6 +23,13 @@ void Scene::draw() {
 	}
 
 	m_choreographer.update();
+
+	for (int i = (int)(round(std::clamp(cfg[Cfg::TrailLength], 1.0f, cfg[Cfg::MaxTrailCount] / cfg[Cfg::SpriteCount]) - 1) * max(round(cfg[Cfg::TrailSpace]), 1)) - 1; i >= 0; i -= (int) max(round(cfg[Cfg::TrailSpace]), 1)) {
```

Comment:
Instead of rendering the trails here, we might want to do it inside the `Sprite`'s `draw` function. Then, instead of doing this huge for loop, we can just do:
```c++
for (const Trail &trail : m_trail) {
    trail.draw(ctx);
}
```
This also means `m_trail` can be protected instead of public.

Reply:
that's probably reasonable, the main reason I did it this way is sprite layering

Reply:
Shit I hadn't considered that. I think we could still do something cleaner by introducing a `draw_trail` function for the `Sprite` though, that would be another way to get rid of this for loop while keeping the trail protected.